tests/store/test_sqlalchemy_store.py

```diff
@@ -317,7 +321,10 @@ def test_log_metric(self):
 
         run = self.store.get_run(run.run_uuid)
 
-        self.assertEqual(4, len(run.data.metrics))
+        sql_run_metrics = self.store._get_run(run.info.run_uuid, ViewType.ALL).metrics
```

Comment:
Btw, could we document what the parameter to `_get_run` is? It seems a bit weird that I ask for a run by id, but also must pass in the state I expect it to be in. Maybe it could default to ViewType.ALL, unless this is a safety thing?

The error message isn't great, btw, if ViewType mismatches -- it would just say `No runs with id <> exists`. Not only is this grammatically incorrect (😛), but it doesn't indicate why there was a problem. Having the caller enforce the ViewType constraint where appropriate might be clearer.

Reply:
That is a good point--I think the motivation was to push down the filter for ACTIVE vs DELETED runs. Maybe the method should return the run and caller should print appropriate message.

Will handle this as a follow on change